src/maliput/routing/derive_lane_s_routes.cc

```diff
@@ -42,48 +42,46 @@ namespace routing {
 namespace {
 
 // Returns true iff @p lane is in @p set.
-bool LaneExistsInSet(const api::LaneEndSet* set, const api::Lane* lane) {
-  MALIPUT_DEMAND(set != nullptr);
+bool LaneExistsInSet(const api::LaneEndSet* set, const api::Lane& lane) {
+  MALIPUT_THROW_UNLESS(set != nullptr);
   for (int i = 0; i < set->size(); ++i) {
-    if (set->get(i).lane == lane) {
+    if (set->get(i).lane == &lane) {
       return true;
     }
   }
   return false;
 }
 
+}  // namespace
+
 // Returns the S coordinate in @p lane that is on the border with @p next_lane.
```

Comment:
Can omit this comment since developers can refer to the docstring in the .h file and effort is needed to keep the two docstrings consistent if both exist.

Reply:
Fixed in [e674557](https://github.com/maliput/maliput/pull/619/commits/e674557d805baf0219c485de6f242853285f2ca4)